### PR TITLE
[do not merge] haskell.nix build setup

### PR DESCRIPTION
**Pull request by jkachmar**
tl;dr I wanted to poke around with something while I was bored; I don't recommend that we merge this at all, but I decided to push it up in case anyone else was interested in taking a look.

---

This is more of an experiment than anything else; I don't think it's something that we'll necessarily ever want to merge, but I wanted to see how difficult it would be to get something up and running with IOHK's [`haskell.nix`](https://github.com/input-output-hk/haskell.nix) build framework.

---

There are a bunch of caveats here, but the quickstart is basically:

1. [Install Nix](https://nixos.org/download.html)
2. [Set up the IOHK binary cache](https://input-output-hk.github.io/haskell.nix/tutorials/getting-started/#setting-up-the-binary-cache)
3. Clone this repo and checkout `jkachmar/haskell-nix`
4. Enter the development shell with either `nix develop` or `nix-shell` (depending on whether your Nix installation supports Flakes or not)

That'll probably take a little while to build, but when it's finished it should drop you into a sandbox with GHC 8.10.5, `cabal-install-3.4.0.0`, `haskell-language-server-1.3.0.0`, and a GHC package DB loaded up with all of the project's dependencies (this allows someone to drop into a REPL with `ghci` and have access to all of the libraries instantly).

---

The value proposition here is threefold:
1. The project can now be built with `nix build .#pg-client:lib:pg-client` in a way that's fully hermetic, down to the C libraries
2. CI builds can be much more aggressively cached, and these caches can be shared with developers
    - i.e. CI can upload cached artifacts for all of the dependencies, so in an ideal scenario no one should have to rebuild anything on a fresh checkout
3. The cached development environment from (2) can include all of the project's development dependencies, including a working Hoogle server
    - There's currently a small bug, but it's better than nothing (cf. https://github.com/input-output-hk/haskell.nix/issues/590#issuecomment-712702992)

### CODE REVIEW

File: pg-client.cabal [hunk 1]
```diff
@@ -24,6 +24,9 @@ custom-setup
     , Cabal  >=1.10
 
 library
+  extra-libraries:
+      krb5
+    , ssl
```
Comment (jkachmar) [inline pg-client.cabal:29]: Should we cherry pick this off into its own PR? These C libraries are required, and `haskell.nix` uses this stanza to figure out what C libs to automatically pull in from nixpkgs, but it seems like we should be calling them out here regardless.

File: Setup.hs [hunk 2]
```diff
@@ -44,8 +44,8 @@ psqlBuildInfo lbi = do
 
   -- On Debian, --includedir and --pkgincludedir return the path
   -- "/usr/include/postgresql" but on Arch Linux, they are respectively
-  -- "/usr/inculde" and "/usr/include/postgresql".  We need to have
-  -- "/usr/inculde/postgresql" as part of our includes for "internal/postgres_fe.h"
+  -- "/usr/include" and "/usr/include/postgresql".  We need to have
+  -- "/usr/include/postgresql" as part of our includes for "internal/postgres_fe.h"
```
Comment (jkachmar) [inline Setup.hs:48]: This could probably be cherry picked out to its own PR as well.
